File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        
        int result = INT_MAX, window = 0;
        for(int r=0, l=0; r<nums.size(); r++){
            
            window += nums[r];
            while(window >= target){
                result = min(result, r - l + 1);
                window -= nums[l];
                l++;
            }
        }
        return result == INT_MAX? 0: result;
    }
};

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        l = 0
        run = 0
        result = len(nums) + 1
        for r in range(len(nums)):
            run += nums[r]
            while run >= target:
                result = min(result, r - l + 1)
                run -= nums[l]
                l += 1
        return 0 if result == len(nums) + 1 else result
                